### DISCUSSION TITLE
`std::optional` support with `NLOHMANN_DEFINE_TYPE_INTRUSIVE`

### DISCUSSION THREAD

**Discussion by hacker-cb** [nlohmann/json#4151] [Ideas]
Hello!

Currently`std::optional` is not supported with `NLOHMANN_DEFINE_TYPE_INTRUSIVE`

Is there any roadmap to make it work without implementing custom `to_json`/`from_json` methods? 

The next code produce compile error now:
```
struct NyClass {
    std::optional<std::string> baseValue;
    NLOHMANN_DEFINE_TYPE_INTRUSIVE(BaseClass, baseValue)
};

```

**Comment by nlohmann** [comment]
You have to define the conversion functions explicitly, see https://json.nlohmann.me/features/arbitrary_types/#how-do-i-convert-third-party-types - there is an example for boost::optional that you can adapt.

**Comment by hacker-cb** [comment]
Thank you for the fast response. I will try.
